gpm_api/tests/conftest.py

```diff
@@ -0,0 +1,322 @@
+import pytest
+import datetime
+from typing import Any, List, Dict, Tuple
+from gpm_api.io.products import get_info_dict, available_products
+import posixpath as pxp
+import ntpath as ntp
+import gpm_api.configs
+from unittest.mock import patch
+
+
+@pytest.fixture(scope="session", autouse=True)
+def mock_configuration():
+    """Patch the user configuration for entire session
+
+    Doing this will retrieve the configuration from pytest memory and not
+    alter the local configuration in ~/.config_gpm_api.yml
+    """
+
+    mocked_configuration = {
```

Comment:
Here we could potentially ask to the PPS guys if we could have an account (mail) that could be used for really testing the connection ...

Reply:
I think in the case we are live testing a server, it moves beyond unit tests and into the integration or contract testing space which can become somewhat redundant if we don't control the server. Unit tests should all be independent from all other functions of the library and operation, hence the large amount of mocking that has had to happen in the `gpm_api.io` module.